scale_build/image/update.py

```diff
@@ -163,9 +163,144 @@ def post_rootfs_setup():
         os.chmod(pkg_mgmt_disabled_path, old_mode | executable_flag)
 
 
+def download_and_install_deb_package(package_name, download_url, deb_filename, post_install_commands=None):
+    """
+    Download and install a .deb package from a URL.
+
+    Args:
+        package_name: Name of the package for logging
+        download_url: Full URL to download the package from
+        deb_filename: Filename of the .deb package
+        post_install_commands: Optional list of additional commands to run in chroot after installation
+    """
+    # Create a temporary directory for the download
+    with tempfile.TemporaryDirectory() as tmpdir:
+        deb_path = os.path.join(tmpdir, deb_filename)
+
+        logger.info(f'Downloading {package_name} from {download_url}')
+
+        # Download the package using curl
+        download_cmd = [
+            'curl',
+            '-L',  # Follow redirects
+            '-o', deb_path,
+            download_url
+        ]
+
+        try:
+            run(download_cmd)
+        except Exception as e:
+            logger.error(f'Failed to download {package_name}: {e}')
+            raise RuntimeError(f'Failed to download {package_name} from {download_url}: {e}')
+
+        # Verify the downloaded file exists and has content
+        if not os.path.exists(deb_path) or os.path.getsize(deb_path) == 0:
+            raise RuntimeError(f'Downloaded {package_name} package is missing or empty: {deb_path}')
+
+        # Copy the package into the chroot
+        chroot_tmp_path = os.path.join(CHROOT_BASEDIR, 'tmp', deb_filename)
+        shutil.copy2(deb_path, chroot_tmp_path)
+
+        # Install the package in the chroot
+        logger.info(f'Installing {package_name} package')
+        try:
+            run_in_chroot(['dpkg', '-i', f'/tmp/{deb_filename}'])
+            # Fix any dependency issues
+            run_in_chroot(['apt-get', 'install', '-f', '-y'])
+
+            # Run any additional post-install commands
+            if post_install_commands:
+                for cmd in post_install_commands:
+                    run_in_chroot(cmd)
+
+            logger.info(f'Successfully installed {package_name} package')
+        except Exception as e:
+            logger.error(f'Failed to install {package_name}: {e}')
+            raise RuntimeError(f'Failed to install {package_name} package: {e}')
+        finally:
+            # Clean up the package from chroot tmp
+            if os.path.exists(chroot_tmp_path):
+                os.unlink(chroot_tmp_path)
+
+
+def install_truenas_file_manager():
```

Comment:
Should we just put deb files URL in the config? These three methods don't seem to be different from one another. We can then just install all packages from `manifest["external-packages"]` without the hardcoding their names in the code.